align_system/algorithms/alignment_adm_component.py

```diff
@@ -47,3 +49,94 @@ def run(self,
             selected_choice)
 
         return selected_choice, best_sample_idx
+
+
+class MedicalUrgencyAlignmentADMComponent(ADMComponent):
+    def __init__(
+        self,
+        attributes=None
+    ):
+        if attributes is None:
+            attributes = {}
+        self.attributes = attributes
+
+    def run_returns(self):
+        return ('chosen_choice',
+                'best_sample_idx')
+
+    def run(
+        self,
+        attribute_prediction_scores,
+        alignment_target,
+    ):
+        """ Align based on medical urgency/KDMA tradeoff """
+        med_urg_str  = "Medical Urgency"
+        if alignment_target is None:
+            raise RuntimeError(
+                "Assumption violated: `alignment_target` was None"
+            )
+
+        target_kdmas = alignment_target_to_attribute_targets(
+            alignment_target,
+            self.attributes)
+
+        target_kdmas = [dict(t) for t in target_kdmas]
+        if len(target_kdmas) > 1:
+            raise RuntimeError("Multi-kdma alignment has not yet been implemented")
```

Comment:
Nit: there's a `NotImplementedError` exception type that seems slightly more appropriate here (and below)